homeassistant/components/rachio/switch.py

```diff
@@ -509,3 +544,70 @@ async def async_added_to_hass(self) -> None:
                 self.hass, SIGNAL_RACHIO_SCHEDULE_UPDATE, self._async_handle_update
             )
         )
+
+
+class RachioValve(CoordinatorEntity[RachioUpdateCoordinator], SwitchEntity):
+    """Representation of one smart hose timer valve."""
+
+    def __init__(
+        self, person, base, data, coordinator: RachioUpdateCoordinator
+    ) -> None:
+        """Initialize a new smart hose valve."""
+        super().__init__(coordinator)
+        self._person = person
+        self._base = base
+        self.id = data[KEY_ID]
+        self._attr_name = data[KEY_NAME]
+        self._attr_unique_id = f"{self.id}-valve"
+        self._static_attrs = data[KEY_STATE][KEY_REPORTED_STATE]
+        self._attr_is_on = KEY_CURRENT_STATUS in self._static_attrs
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (
+                    DOMAIN_RACHIO,
+                    self._base.serial_number,
```

Comment:
Sorry @bdraco, I just thought of one more thing.. 😟 I think the device identifiers should probably point to the valve itself, not the base station. In the current format you wouldn't be able to tie sensors to the individual valve (like battery status). Thoughts?

Reply:
You can add device info to the sensors with the same identifiers if they are on the same device and they will get linked.  If its a difference device, use the identifier for that device and use `via_device`

Reply:
I guess the question is, is the valve itself the device or is the hub? After thinking about it more, it seems like the valve should be.

Reply:
I think the Hub is a device and the valve is also a device

I'd make the valve connected to the hub using `via_device`

Reply:
If the base has no entities is it a device?

This doesn't seem to work in the switch device info: 
` via_device=(DOMAIN_RACHIO, self._base.serial_number)`

Reply:
> If the base has no entities is it a device?
> 
> This doesn't seem to work in the switch device info: ` via_device=(DOMAIN_RACHIO, self._base.serial_number)`

You would need to create the device manually in that case. Example https://github.com/home-assistant/core/blob/9a863638f6fd7ef1ac30d0c6159eed1be6f39e02/homeassistant/components/isy994/__init__.py#L188

But it may make more sense to not create the hub as a device and link everything valve instead if it will never have any entities